Search test

saracarl opened the pull request.

worleydl committed:
Add Elasticsearch config/gem, init rake task

worleydl committed:
Cleanup logging, add es_reset task

worleydl committed:
Cleanup env config, move ES client to lib module

worleydl committed:
Export method

worleydl committed:
es_reindex task

worleydl committed:
Cleanup id field management for bulk

worleydl committed:
collection/user/work indexing.
Cleanup reindexing code

worleydl committed:
Add collection_id/is_public to page schema

worleydl committed:
Handle null collections

worleydl committed:
More null collection

worleydl committed:
Multithreaded indexing

worleydl committed:
ES search impl for collection level search

worleydl committed:
Cleanup query

worleydl committed:
ES pagination

worleydl committed:
Cleanup manual offset code

worleydl committed:
Map text operators to symbol versions for SQS

worleydl committed:
Fix phrase queries

worleydl committed:
Cleanup SQS replacements

worleydl committed:
Work level query

worleydl committed:
Populate page::work_id for elastic

worleydl committed:
Add docset to schema/indexer

worleydl committed:
ES backing for findaproject

worleydl committed:
Trying out partial setup

worleydl committed:
Tap into tabs style

worleydl committed:
Building out logic for new serp

worleydl committed:
Building out object inflation

worleydl committed:
Fix index filters for cross collection search

worleydl committed:
Early search result rendering

worleydl committed:
Fix issue with inflation

worleydl committed:
Dynamic tabs/page count

worleydl committed:
General weighting for federated searches

worleydl committed:
html safe render

worleydl committed:
Pagination test

worleydl committed:
Hookup pagination

worleydl committed:
Link dev

worleydl committed:
Building out filters

worleydl committed:
Show active tab for filter

worleydl committed:
Fix all filter count

worleydl committed:
Fix issues with initial state

worleydl committed:
Cap pagination at 10k

worleydl committed:
Sanitize instead of using raw HTML for snippets

worleydl committed:
Modify user indexing: only owners and no staff users

worleydl committed:
Link up user results

worleydl committed:
Link up collection results

worleydl committed:
Link up work results

worleydl committed:
Link up page results

worleydl committed:
Setup page partial for search results

worleydl committed:
Simplify page partial

worleydl committed:
Tap into existing highlighting code

worleydl committed:
Setup work partial

worleydl committed:
WIP elastic translations

worleydl committed:
Cleanup escaped collection footer

worleydl committed:
Fix table issue with page partial

worleydl committed:
Exclude works missing a collection from indexing

worleydl committed:
Use to_snippet for user/collection snippets

worleydl committed:
Fold docsets into collections for index/search

worleydl committed:
Hookup docset partial render

worleydl committed:
Early collection access

worleydl committed:
Expand collection access

worleydl committed:
Add docsets to collection filter

worleydl committed:
Expand works indexing in prep for access queries

worleydl committed:
Setup filtering for work type

worleydl committed:
Prep page schema for filters

worleydl committed:
Update page filters

worleydl committed:
Add Elastic delta concern and attach to required models

worleydl committed:
Cleanup lingering issues with work model

worleydl committed:
Fix delta delete

worleydl committed:
Don't setup hooks if elastic isn't enabled

saracarl committed:
all of Dan's changes for ElasticSearch

saracarl committed:
gemfile update

WillNigel23 reviewed (commented):
Excited for this :clap:

worleydl committed:
Improve layout of tabbed search results

worleydl committed:
No results template

worleydl committed:
Exception handling for findaproject query

worleydl committed:
Translations cleanup

worleydl committed:
Reuse query generation

worleydl committed:
Add blocked collections to filters

worleydl committed:
Cleanup dead code

worleydl committed:
Consistent partials in main search results

saracarl committed:
Merge remote-tracking branch 'dans-work/development' into search_test

saracarl committed:
Merge branch 'development' into search_test

worleydl committed:
Prep for delta permission updates

worleydl committed:
Early version of permissions indexer

worleydl committed:
Setup delete by query optimization for bulk page deletes

worleydl committed:
Finish up permission delta indexer

worleydl committed:
Must reindex works on coll/docset permission changes

saracarl committed:
added elasticsearch query logging

benwbrum committed:
Add ES query logging

worleydl committed:
ES: Setup whitespace analyzer for searchable metadata on works

worleydl committed:
ES: Basic query intent for ID searches in work metadata

worleydl committed:
ES: Add more weight to phrase matches on pages

worleydl committed:
ES: Apply ES relevance to search_attempt searches

worleydl committed:
ES: Fold Page updates into delta indexer

worleydl committed:
Cleanup work partial title in findaproject results

worleydl committed:
Make page results consistent with other types

worleydl committed:
Add work breadcrumb on page results

worleydl committed:
Prep for org level search

worleydl committed:
Org search

worleydl committed:
Fix org search type filters

worleydl committed:
Route nested searches to tabbed results

worleydl committed:
Update search heading for different types

worleydl committed:
Improved breadcrumbs for Pages/Works

worleydl committed:
Schema tweaks for improved identifier search

worleydl committed:
Use work style on page partial for consistent margins

worleydl committed:
Handle exceptions for client facing ES requests

worleydl committed:
Basic exception handling for sync task

worleydl committed:
Smaller breadcrumbs, moved to footer.

worleydl committed:
Drop recent activity, sticky search modes

worleydl committed:
Improvements to identifier search

worleydl committed:
Lock search results to viewport width, allow side scroll

worleydl committed:
Don't add params to searchbox if not set

worleydl committed:
WIP alias management

worleydl committed:
Build out environment support

worleydl committed:
Fix index boosts

worleydl committed:
Remove double padding from page results, consistent breadcrumbs

worleydl committed:
Cleanup phrase query generator

worleydl committed:
Merge dev

worleydl committed:
Rollover fixes

saracarl committed:
merge dev in

saracarl committed:
merging dev and dans code

saracarl committed:
more dans code

benwbrum committed:
Checkpoint

benwbrum committed:
Checkpoint on search integration -- work search

benwbrum committed:
Document sets and collection search work

benwbrum committed:
Checkpoint fighting with context-aware breadcrumbs

benwbrum committed:
Merge branch 'development' into search_test

benwbrum committed:
Fix breadcrumbs for search results

worleydl committed:
Schema patches:

- Drop wildcard from identifier analysis as it can cause match-all
  behavior due to drop purewords filter.
- Expand terms with underscores into `title.no_underscores` and use in
  search.

benwbrum committed:
Fix ActiveRecord datatype problem

saracarl committed:
Merge pull request #4539 from worleydl/search_test_patch

Schema patches:

benwbrum committed:
Conditionalize old search vs elastic search

saracarl committed:
remove unused lang key

saracarl committed:
add missing translations

benwbrum committed:
New messages

benwbrum committed:
Merge branch 'search_test' of github.com:benwbrum/fromthepage into search_test

saracarl committed:
translated additional messages

benwbrum committed:
Fighting with romance languages

saracarl committed:
more translations

saracarl committed:
added missing message

benwbrum committed:
Fix test failures
